src/uct/ib/rc/base/rc_iface.c

```diff
@@ -196,10 +196,13 @@ ucs_status_t uct_rc_iface_query(uct_rc_iface_t *iface,
         iface_attr->cap.flags              |= UCT_IFACE_FLAG_ATOMIC_DEVICE;
 
         iface_attr->cap.atomic64.op_flags  |= UCS_BIT(UCT_ATOMIC_OP_ADD);
-        iface_attr->cap.atomic64.fop_flags |= UCS_BIT(UCT_ATOMIC_OP_ADD)  |
-                                              UCS_BIT(UCT_ATOMIC_OP_CSWAP);
+        iface_attr->cap.atomic64.fop_flags |= UCS_BIT(UCT_ATOMIC_OP_ADD);
+#if HAVE_IB_EXT_ATOMICS
+        iface_attr->cap.atomic64.fop_flags |= UCS_BIT(UCT_ATOMIC_OP_CSWAP);
```

Comment:
cswap 64bit is IBTA atomic. can be always supported

Reply:
But it's implemented via EXP:
https://github.com/openucx/ucx/blob/master/src/uct/ib/rc/verbs/rc_verbs_ep.c#L422

Reply:
it's SWAP, not CSWAP